backend/pkg/api/handlers/internal.go

```diff
@@ -293,6 +293,31 @@ func (h *HandlerService) InternalDeleteValidatorDashboard(w http.ResponseWriter,
 	returnNoContent(w)
 }
 
+func (h *HandlerService) InternalPutValidatorDashboardArchiving(w http.ResponseWriter, r *http.Request) {
```

Comment:
This is func is missing checks as to whether unarchiving is allowed. It should go through all perks that could be violated and check against them before unarchiving (may require new data access funcs)

Reply:
Right missed that, added handling for this.

But while thinking about permissions, I noticed that the other way should also be checked - if a dashboard is archived, basically no VDB endpoint should return any actual data, right? Unless I'm missing what "dashboard archival" is even about.

So in order to handle this without copy pasting the same archival status handling 20+ times for all endpoints, I also added a suggestion by messing with the `GetVDBAuthMiddleware` (hopefully not breaking staging until implemented by data access). If that's not clean enough we could move the `/archival` endpoint to another route, but lmk what you think